src/app.py

```
@@ -137,89 +143,107 @@ def _parse_buttons(message: types.Message) -> Optional[List[List[MessageButton]]
-                    # However, for simplicity, let's try direct click first.
-                    # If issues arise, one would do:
-                    # if current_client is not client: # i.e., it's a temporary client
-                    #    message = await current_client.get_messages(await current_client.get_input_entity(req.bot_username), ids=message.id)
-
-                    await message.click(text=req.button_text, data=req.callback_data)
-                except Exception as e: # Catches errors specifically from message.click()
+                    # Click the button on the fetched message
+                    # message_to_click is bound to current_client which is used for the conversation
+                    await message_to_click.click(text=req.button_text, data=req.callback_data)
+                except Exception as e: 
                     raise HTTPException(status_code=400, detail=f"Failed to press button: {e}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)